driver-core/src/main/com/mongodb/event/CommandSucceededEvent.java

```diff
@@ -36,31 +36,54 @@ public final class CommandSucceededEvent extends CommandEvent {
 
     /**
      * Construct an instance.
+     *
      * @param requestContext the request context
+     * @param operationId the operation id
      * @param requestId the request id
      * @param connectionDescription the connection description
      * @param commandName the command name
      * @param response the command response
      * @param elapsedTimeNanos the non-negative elapsed time in nanoseconds for the operation to complete
-     * @since 4.4
+     * @since 4.10
      */
-    public CommandSucceededEvent(@Nullable final RequestContext requestContext, final int requestId,
+    public CommandSucceededEvent(@Nullable final RequestContext requestContext, final long operationId, final int requestId,
             final ConnectionDescription connectionDescription, final String commandName, final BsonDocument response,
             final long elapsedTimeNanos) {
-        super(requestContext, requestId, connectionDescription, commandName);
+        super(requestContext, operationId, requestId, connectionDescription, commandName);
         this.response = response;
         isTrueArgument("elapsed time is not negative", elapsedTimeNanos >= 0);
         this.elapsedTimeNanos = elapsedTimeNanos;
     }
 
+    /**
+     * Construct an instance.
+     * @param requestContext the request context
+     * @param requestId the request id
+     * @param connectionDescription the connection description
+     * @param commandName the command name
+     * @param response the command response
+     * @param elapsedTimeNanos the non-negative elapsed time in nanoseconds for the operation to complete
+     * @since 4.4
+     * @deprecated Prefer
+     * {@link CommandSucceededEvent#CommandSucceededEvent(RequestContext, long, int, ConnectionDescription, String, BsonDocument, long)}
+     */
+    @Deprecated
```

Comment:
Why deprecate these, rather than just leave them as helpers? In what way is the `requestContext` optional in a way that `operationId` is not?

Reply:
So we can minimize API surface over time by removing these in the next major release.

The `requestContext` is nullable because it is optionally configured via `MongoClientSettings#getContextProvider()`.  The `operationId` is really required now, so we don't want to keep around constructors that don't provide a value for it.